dashboard/app/models/pd/application/facilitator1920_application.rb

```diff
@@ -117,6 +130,18 @@ def log_status
       status_log.push({status: status, at: Time.zone.now})
     end
 
+    def lock!
+      super
+
+      # delete any unsent emails, and queue a new status email if appropriate
+      emails.unsent.destroy_all
+      queue_email(status) if should_send_decision_email?
```

Comment:
A problem here is that when changing status and locking an application in the same request, this function, as called by the controller, will be examining the previous status and not the new incoming one.

Reply:
ooh good point! Does our UI allow this? (we should fix either way, but curious if this could be the cause of issues we've been seeing or if it's just a good idea to clean up)